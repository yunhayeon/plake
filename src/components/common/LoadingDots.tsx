import clsx from "clsx";

interface LoadingDotsProps {
  color?: "gray" | "purple";
}

const colorClasses = {
  gray: "bg-gray-200",
  purple: "bg-purple-300",
} as const;

const LoadingDots = ({ color = "gray" }: LoadingDotsProps) => {
  return (
    <div className="flex h-full w-full flex-grow items-center justify-center">
      <div className="z-50 flex space-x-3.5" role="status" aria-live="polite">
        {[...Array(3)].map((_, i) => (
          <div
            key={`dot-${i}`}
            className={clsx(
              "h-3 w-3 animate-bounce rounded-full",
              colorClasses[color],
            )}
            style={{ animationDelay: `${i * 0.15}s` }}
          />
        ))}
        <span className="sr-only">Loading...</span>
      </div>
    </div>
  );
};

export default LoadingDots;
