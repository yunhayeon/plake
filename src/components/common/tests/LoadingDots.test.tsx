import { render } from "@testing-library/react";

import LoadingDots from "@/components/common/LoadingDots";

describe("LoadingDots 컴포넌트 테스트", () => {
  it("기본 color(gray)로 렌더링된 결과가 스냅샷과 일치한다", () => {
    const { container } = render(<LoadingDots />);
    expect(container).toMatchSnapshot();
  });

  it("color가 purple일 때 스냅샷과 일치한다", () => {
    const { container } = render(<LoadingDots color="purple" />);
    expect(container).toMatchSnapshot();
  });
});
