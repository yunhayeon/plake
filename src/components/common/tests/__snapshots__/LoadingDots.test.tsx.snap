// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingDots 컴포넌트 테스트 color가 purple일 때 스냅샷과 일치한다 1`] = `
<div>
  <div
    class="flex h-full w-full flex-grow items-center justify-center"
  >
    <div
      aria-live="polite"
      class="z-50 flex space-x-3.5"
      role="status"
    >
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-purple-300"
        style="animation-delay: 0s;"
      />
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-purple-300"
        style="animation-delay: 0.15s;"
      />
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-purple-300"
        style="animation-delay: 0.3s;"
      />
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;

exports[`LoadingDots 컴포넌트 테스트 기본 color(gray)로 렌더링된 결과가 스냅샷과 일치한다 1`] = `
<div>
  <div
    class="flex h-full w-full flex-grow items-center justify-center"
  >
    <div
      aria-live="polite"
      class="z-50 flex space-x-3.5"
      role="status"
    >
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-gray-200"
        style="animation-delay: 0s;"
      />
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-gray-200"
        style="animation-delay: 0.15s;"
      />
      <div
        class="h-3 w-3 animate-bounce rounded-full bg-gray-200"
        style="animation-delay: 0.3s;"
      />
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
